tests/ast/decoder.py

```diff
@@ -614,6 +616,34 @@ def decode_expr(self, expr: proto.Expr) -> Any:
                 col_name = expr.sp_dataframe_col.col_name
                 df = self.decode_expr(expr.sp_dataframe_col.df)
                 return df[col_name]
+            case "sp_dataframe_collect":
+                df = self.symbol_table[expr.sp_dataframe_collect.id.bitfield1][1]
+                d = MessageToDict(expr.sp_dataframe_collect)
+                statement_params = {}
+                statement_params_list = d.get("statementParams", [])
+                if len(statement_params_list) > 0:
+                    statement_params_list_map = statement_params_list[0]
+                    statement_params[
+                        statement_params_list_map["1"]
+                    ] = statement_params_list_map["2"]
+                log_on_exception = d.get("logOnException", False)
```

Comment:
I think it's better to have logic here that only extracts the values for `block`, `log_on_exception`, and `case_sensitive` and not give them default values if not present since they all should always be populated -- if you look at the snowpark API there are different overloads with different default values.

It's okay to have a default value for `no_wait`.

Reply:
Done - also fixed to handle multiple entries in map...